blockchain/chaingen/generator.go

```diff
@@ -78,15 +78,25 @@ func (s *SpendableOut) Amount() dcrutil.Amount {
 	return s.amount
 }
 
-// MakeSpendableOutForTx returns a spendable output for the given transaction
-// block height, transaction index within the block, and transaction output
-// index within the transaction.
+// MakeSpendableOutForTx returns a spendable output for a regular transaction
 func MakeSpendableOutForTx(tx *wire.MsgTx, blockHeight, txIndex, txOutIndex uint32) SpendableOut {
+	return MakeSpendableOutForTxInternal(tx, blockHeight, txIndex, txOutIndex, wire.TxTreeRegular)
+}
+
+// MakeSpendableOutForSTx returns a spendable output for a stake transaction
+func MakeSpendableOutForSTx(tx *wire.MsgTx, blockHeight, txIndex, txOutIndex uint32) SpendableOut {
+	return MakeSpendableOutForTxInternal(tx, blockHeight, txIndex, txOutIndex, wire.TxTreeStake)
+}
+
+// MakeSpendableOutForTx returns a spendable output for the given transaction
+// block height, transaction index within the block, transaction tree and
+// transaction output index within the transaction.
+func MakeSpendableOutForTxInternal(tx *wire.MsgTx, blockHeight, txIndex, txOutIndex uint32, tree int8) SpendableOut {
```

Comment:
This should be unexported since it's internal. Also, please define it before it's used to be consistent.